src/base.ts

```diff
@@ -41,76 +30,69 @@ const withPagination = <T>(json: {
   return { content, fieldsStats, pagination, topListing }
 }
 
-export const resolveServiceUrl = (service: Service): string => {
-  let url: string
-  switch (service) {
-    case Service.CAR:
-      url = apiClient.configuration.carServiceUrl
-      break
-    case Service.CATALOGUE:
-      url = apiClient.configuration.catalogueServiceUrl
-      break
-    case Service.SEARCH:
-      url = apiClient.configuration.searchServiceUrl
-      break
-    case Service.DEALER:
-      url = apiClient.configuration.dealerServiceUrl
-      break
-    case Service.OPTION:
-      url = apiClient.configuration.optionServiceUrl
-      break
-    case Service.ANALYTICS:
-      url = apiClient.configuration.analyticsServiceUrl
-      break
-    case Service.USER_NOTIFICATION:
-      url = apiClient.configuration.userNotificationServiceUrl
-      break
-    case Service.TOKEN_REFRESH:
-      url = apiClient.configuration.tokenRefreshServiceUrl
-      break
-    default:
-      throw new Error(`Tried to resolve url of unknown service "${service}"`)
+const authorizationHeader = () => {
+  if (!apiClient.tokens.accessToken) {
+    return null
   }
 
-  if ((url || "").trim()) {
-    return url
-  }
+  return `Bearer ${apiClient.tokens.accessToken}`
+}
 
-  throw new Error(`Missing endpoint configuration for "${service}" service`)
+export interface RequestOptions {
+  headers?: Record<string, string>
+  // TODO: think about this naming
+  serviceUrl?: string
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+export interface RequestOptionsWithRecaptcha extends RequestOptions {
+  recaptchaToken?: string
+}
+
+type Method = "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+
+interface RequestOptionsWithMethod extends RequestOptions {
+  method?: Method
+}
+
+export const getHost = ({ serviceUrl }) => {
+  const url = serviceUrl || apiClient.configuration.apiGatewayUrl
+
+  if (!url) {
+    throw new Error(
+      'ApiClient not configured, please run: ApiClient.configure({ apiGatewayUrl: "your.api.gateway" }'
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return url
 }
 
-export const fetchPath = async (
-  service: Service,
-  path: string,
-  options = {}
-) => {
-  const prefix = resolveServiceUrl(service)
-  const url = `${prefix}/${stripLeadingSlash(path)}`
+export const fetchPath = async ({
+  path,
+  body,
+  options,
+}: {
+  path: string
+  body?: string
+  options: RequestOptionsWithMethod
```

Comment:
1. I really liked to have path as the first (unnamed) argument as it's the only required one. the majority of simple GET would be a tad more straight forward

2. would options be optional?

Reply:
1. This is in line with what @lkappeler is doing for token handling change (https://github.com/carforyou/carforyou-api-client-pkg/pull/175/files#diff-440b55a9ff9e0486020fc49197f3278d14446306fc0c5d5107b48139c20bfda6R95). I'd like to keep it consistent to minimize the conflicts.

2. I thought about it and opted not to make it optional for the internal methods. My reasoning is that this way we ensure (via type safety) that we include and pass options for the new API calls down to wrappers around fetch since new logic of overwriting gateway URL relies on that.